src/components/shared/CodeViewer/CodeViewer.tsx

```diff
@@ -1,165 +1,92 @@
-          />
-        </div>
-      ),
-    [code, language],
-  );
-
-  return (
-    <div className="border rounded-md h-full bg-slate-900 flex flex-col">
-      <div className="flex justify-between items-center p-2 sticky top-0 z-10 bg-slate-800">
-        <h3 className="text-secondary font-medium ml-2">
-          {filename} <span className="text-sm">(Read Only)</span>
-        </h3>
-        <Button
-          variant="ghost"
-          size="icon"
-          onClick={handleEnterFullscreen}
-          className="text-gray-300 hover:text-slate-800"
-          title="View fullscreen"
-        >
-          <Maximize2 className="size-4" />
-        </Button>
-      </div>
-      <div className="flex-1 relative">{syntaxHighlighter}</div>
-    </div>
-  );
-};
-
-/*
- * A basic Window/Virtualizer implementation for large code files
- * It will only render the lines of code that the user is looking at, rather than the entire file
- * */
-const DEFAULT_HEIGHT = 128;
-const WindowedCodeViewer = ({
-  code,
-  language = "yaml",
-  fontSize = "0.75rem",
-}: CodeViewerProps & { fontSize?: string }) => {
-  const lines = useMemo(() => code.split("\n"), [code]);
-  const totalLines = lines.length;
-  const scrollRef = useRef<HTMLDivElement>(null);
-  const [containerHeight, setContainerHeight] = useState(DEFAULT_HEIGHT);
-  const [scrollTop, setScrollTop] = useState(0);
-
-  // Measure container height using ResizeObserver
-  useEffect(() => {
-    const node = scrollRef.current;
-    if (!node) return;
-    const handleResize = () =>
-      setContainerHeight(
-        node.clientHeight > 0 ? node.clientHeight : DEFAULT_HEIGHT,
-      );
-
-    handleResize();
-    let observer: ResizeObserver | null = null;
-    if (window.ResizeObserver) {
-      observer = new ResizeObserver(handleResize);
-      observer.observe(node);
-    } else {
-      window.addEventListener("resize", handleResize);
+  const dynamicHeight = useMemo(() => {
+    if (isFullscreen) {
+      return "calc(100vh - 100px)"; // Fullscreen height
     }
-    return () => {
-      if (observer) observer.disconnect();
-      else window.removeEventListener("resize", handleResize);
-    };
-  }, []);
-
-  const lineHeight = useMemo(() => getLineHeight(fontSize), [fontSize]);
 
-  const visibleLines = Math.max(1, Math.ceil(containerHeight / lineHeight));
-  const overscan = 2;
+    if (height) {
+      return height;
+    }
 
-  // Calculate which lines to render
-  let startLine = Math.floor(scrollTop / lineHeight) - overscan;
-  startLine = Math.max(0, startLine);
-  let endLine = startLine + visibleLines + overscan * 2;
-  endLine = Math.min(totalLines, endLine);
+    const lines = code.split("\n").length;
+    const lineHeight = 19;
+    const headerHeight = 40;
+    const padding = 20;
 
-  const offsetY = startLine * lineHeight;
-  const visibleCode = lines.slice(startLine, endLine).join("\n");
+    return `${Math.max(lines * lineHeight + headerHeight + padding, 50)}px`;
```

Comment:
For tidiness can we make this calculation a separate variable?

e.g.
```suggestion
    const dynamicHeight = Math.max(lines * lineHeight + headerHeight + padding, 50);
    return `${dynamicHeight}px`;
```